src/js/BookReader.js

```diff
@@ -889,12 +889,14 @@ BookReader.prototype.drawLeafsThumbnail = function(seekIndex) {
 
         const img = document.createElement("img");
         const thumbReduce = floor(book.getPageWidth(leaf) / this.thumbWidth);
-
+        // use prefetched img src first if previously requested & available img is good enough
+        const prefetchedImg = this.prefetchedImgs[leaf] || {};
+        const imageURI = prefetchedImg.reduce <= thumbReduce ? prefetchedImg.uri : this._getPageURI(leaf, thumbReduce);
```

Comment:
Thumbnail: added check of prefetchedImgs if img has been fetched && if its reduction factor is better than the thumbnail one. If so, then we reuse to save a request.

Reply:
BLOCKER? If `prefetchedImg` is `{}`, then `prefetchedImg.reduce` in the next line will be accessing a non-existent property, no?

Reply:
@bfalling If  `prefetchedImg` is `{}` and `prefetchedImg.reduce` is undefined, the comparison evals to false (since all greater/less than comparisons with `undefined` are false) and we end up in the second branch of the ternary.